bindings/nodejs/src/client.rs

```diff
@@ -6,21 +6,26 @@ use std::sync::Arc;
 use iota_sdk_bindings_core::{
     call_client_method as rust_call_client_method,
     iota_sdk::client::{mqtt::Topic, Client, ClientBuilder},
-    listen_mqtt as rust_listen_mqtt, ClientMethod, Response,
+    listen_mqtt as rust_listen_mqtt, ClientMethod, Response, Result,
 };
 use neon::prelude::*;
+use tokio::sync::RwLock;
 
 type JsCallback = Root<JsFunction<JsObject>>;
 
+// Wrapper so we can destroy the ClientMethodHandler
+pub type ClientMethodHandlerWrapperInner = Arc<RwLock<Option<ClientMethodHandler>>>;
+// Wrapper because we can't impl Finalize on ClientMethodHandlerWrapperInner
+pub struct ClientMethodHandlerWrapper(pub ClientMethodHandlerWrapperInner);
 pub struct ClientMethodHandler {
```

Comment:
This kind of pattern can usually be done with 2 types, why do we need 3 🤔 ?

Reply:
Inner is a type not a struct. i didnt try rewriting it though, this is how it is in wallet

Reply:
I guess I was confused because usually it would be the other way around. The type actually being used (here `ClientMethodHandlerWrapper`) should not have any suffix. So I'd say `ClientMethodHandler` should be `ClientMethodHandlerInner`, `ClientMethodHandlerWrapper` should be `ClientMethodHandler` and `ClientMethodHandlerWrapperInner` something else or not even exist.

Reply:
Sure we can open a ticket for it

Reply:
Definitely feels like the type def isn't needed here.